### ISSUE TITLE
top level await

### ISSUE THREAD

**Issue by jimmywarting** [jsfiddle/jsfiddle-issues#1674]
Ever thought about creating a iife async function so that you could use top level await?

would be cool!

**Comment by jimmywarting** [comment]
still want to be able to just simply import scripts in a easier way. top level await would be awesome to have

**Comment by jimmywarting** [comment]
still want to be able to just simply import scripts in a easier way. top level await would be awesome to have

**Comment by jimmywarting** [comment]
still want to be able to just simply import scripts in a easier way. top level await would be awesome to have

**Comment by jimmywarting** [comment]
still want to be able to just simply import scripts in a easier way. top level await would be awesome to have

**Comment by jimmywarting** [comment]
FINE I'M CLOSING THIS MYSELF!

**Comment by KaKi87** [comment]
Hello,
Could you please reopen this ?
Thanks.

**Comment by mattrossman** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#browser_compatibility

<img width="500" alt="browser compatibility table for top level await showing green checks across the board" src="https://user-images.githubusercontent.com/22670878/229023995-d33f97a8-703e-4762-a5dd-1dfa0d62323c.png">

Top level await is supported in all modern browsers, could JSFiddle please be updated to remove the error indicator from the editor?

<img width="410" alt="error in editor, missing semicolon, though no such error is present" src="https://user-images.githubusercontent.com/22670878/229024798-22e30566-fb7e-449a-a4c6-9ed1fffc7e41.png">

**Comment by mattrossman** [comment]
@oskarkrawczyk 
That syntax is top level await, it's supported in all browsers since 2021 https://github.com/jsfiddle/jsfiddle-issues/issues/1674#issuecomment-1491281331.
I'm confused what was fixed here, as this issue was requesting that syntax 😅


The particular screenshot I showed was using Three.js to load a glTF asset, but you can await any promise at top level
```js
await Promise.resolve()
```

JSFiddle can execute this code, it just displays an error in the editor.

https://jsfiddle.net/mattrossman/78gbLzd5/1/

_Note: the `import` from lodash is just so that JSFiddle treats this script as `type="module"`, though it would be nice if we had a better way to trigger this._ 

   Editor | Console
---|---
<img width="300" src="https://github.com/jsfiddle/jsfiddle-issues/assets/22670878/aca475b9-b102-4959-b950-e78131126234" /> | <img width="300" src="https://github.com/jsfiddle/jsfiddle-issues/assets/22670878/b331cd25-ab34-4894-8cca-b2b953cdcf4b" />




In a fresh project VSCode doesn't have an issue:
<img width="300" src="https://github.com/jsfiddle/jsfiddle-issues/assets/22670878/7338a705-074d-4d2c-93fd-16ebbc893988" />



ESLint accepts top level await when configured with ECMAScript 2022 or newer.
[(ESLint Playground)](https://eslint.org/play/#eyJ0ZXh0IjoiYXdhaXQgUHJvbWlzZS5yZXNvbHZlKCkiLCJvcHRpb25zIjp7InJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjpbImVycm9yIl0sImZvci1kaXJlY3Rpb24iOlsiZXJyb3IiXSwiZ2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjpbImVycm9yIl0sIm5vLWNhc2UtZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWNsYXNzLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb21wYXJlLW5lZy16ZXJvIjpbImVycm9yIl0sIm5vLWNvbmQtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1iaW5hcnktZXhwcmVzc2lvbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1jb25kaXRpb24iOlsiZXJyb3IiXSwibm8tY29udHJvbC1yZWdleCI6WyJlcnJvciJdLCJuby1kZWJ1Z2dlciI6WyJlcnJvciJdLCJuby1kZWxldGUtdmFyIjpbImVycm9yIl0sIm5vLWR1cGUtYXJncyI6WyJlcnJvciJdLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tZHVwZS1lbHNlLWlmIjpbImVycm9yIl0sIm5vLWR1cGUta2V5cyI6WyJlcnJvciJdLCJuby1kdXBsaWNhdGUtY2FzZSI6WyJlcnJvciJdLCJuby1lbXB0eSI6WyJlcnJvciJdLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tZW1wdHktcGF0dGVybiI6WyJlcnJvciJdLCJuby1lbXB0eS1zdGF0aWMtYmxvY2siOlsiZXJyb3IiXSwibm8tZXgtYXNzaWduIjpbImVycm9yIl0sIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6WyJlcnJvciJdLCJuby1mYWxsdGhyb3VnaCI6WyJlcnJvciJdLCJuby1mdW5jLWFzc2lnbiI6WyJlcnJvciJdLCJuby1nbG9iYWwtYXNzaWduIjpbImVycm9yIl0sIm5vLWltcG9ydC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW52YWxpZC1yZWdleHAiOlsiZXJyb3IiXSwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOlsiZXJyb3IiXSwibm8tbG9zcy1vZi1wcmVjaXNpb24iOlsiZXJyb3IiXSwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tbmV3LW5hdGl2ZS1ub25jb25zdHJ1Y3RvciI6WyJlcnJvciJdLCJuby1ub25vY3RhbC1kZWNpbWFsLWVzY2FwZSI6WyJlcnJvciJdLCJuby1vYmotY2FsbHMiOlsiZXJyb3IiXSwibm8tb2N0YWwiOlsiZXJyb3IiXSwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjpbImVycm9yIl0sIm5vLXJlZGVjbGFyZSI6WyJlcnJvciJdLCJuby1yZWdleC1zcGFjZXMiOlsiZXJyb3IiXSwibm8tc2VsZi1hc3NpZ24iOlsiZXJyb3IiXSwibm8tc2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6WyJlcnJvciJdLCJuby1zcGFyc2UtYXJyYXlzIjpbImVycm9yIl0sIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjpbImVycm9yIl0sIm5vLXVuZGVmIjpbImVycm9yIl0sIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjpbImVycm9yIl0sIm5vLXVucmVhY2hhYmxlIjpbImVycm9yIl0sIm5vLXVuc2FmZS1maW5hbGx5IjpbImVycm9yIl0sIm5vLXVuc2FmZS1uZWdhdGlvbiI6WyJlcnJvciJdLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOlsiZXJyb3IiXSwibm8tdW51c2VkLWxhYmVscyI6WyJlcnJvciJdLCJuby11bnVzZWQtcHJpdmF0ZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLXVudXNlZC12YXJzIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicmVxdWlyZS15aWVsZCI6WyJlcnJvciJdLCJ1c2UtaXNuYW4iOlsiZXJyb3IiXSwidmFsaWQtdHlwZW9mIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fSwiZWNtYVZlcnNpb24iOjIwMjIsInNvdXJjZVR5cGUiOiJtb2R1bGUifX19).

However, I don't see an ESLint error in your screenshot—that looks like an error from VSCode's TypeScript language server given the `ts(1308)` tag. TypeScript accepts top-level await with "module" set to ES2022 or later.
[(TypeScript Playground)](https://www.typescriptlang.org/play/?module=7#code/IYd2EsBcAIAUCcD2BbcBnApgOnhtiAbANwwAoBKAKEowA8AHReGAbwF8g)

**Comment by mattrossman** [comment]
Yes, an editor option to treat the script as a module would be great!

**Comment by oskarkrawczyk** [comment]
No longer relevant since moving to Monaco for the editors
